### PR TITLE
More documentation fixes

### PR DESCRIPTION
**Pull request by pepone**
This PR adds a number of missing doc comments, there is still more missing.

### CODE REVIEW

File: csharp/src/Ice/IncomingFrame.cs [hunk 1]
```diff
@@ -145,6 +147,10 @@ public void DecompressPayload()
             }
         }
 
+        /// <summary>Constructs a new <see cref="IncomingFrame"/>.</summary>
+        /// <param name="data">The frame data.</param>
+        /// <param name="protocol">The frame protocol.</param>
+        /// <param name="sizeMax">The maximum payload size, checked during decompress.</param>
```
Comment (externl) [inline csharp/src/Ice/IncomingFrame.cs:153]:
```suggestion
        /// <param name="sizeMax">The maximum payload size, checked during decompression.</param>
```

File: csharp/src/Ice/OutgoingRequest.cs [hunk 2]
```diff
@@ -9,7 +9,8 @@
 
 namespace ZeroC.Ice
 {
-    // publicly visible Ice-internal class used as base class for requests
+    /// <summary>publicly visible Ice-internal class used as base class for requests.</summary>
```
Comment (externl) [inline csharp/src/Ice/OutgoingRequest.cs:12]:
```suggestion
    /// <summary>Publicly visible Ice-internal class used as base class for requests.</summary>
```

File: csharp/src/Ice/OutgoingRequest.cs [hunk 3]
```diff
@@ -45,13 +46,18 @@ private protected OutgoingRequest(InputStreamReader<TReturnValue> reader, bool c
         }
     }
 
-    // publicly visible Ice-internal class used to make request without input parameters.
+    /// <summary>publicly visible Ice-internal class used to make request without input parameters.</summary>
```
Comment (externl) [inline csharp/src/Ice/OutgoingRequest.cs:49]:
```suggestion
    /// <summary>Publicly visible Ice-internal class used to make request without input parameters.</summary>
```

File: csharp/src/Ice/OutgoingRequest.cs [hunk 4]
```diff
@@ -77,9 +94,11 @@ public sealed class OutgoingRequestWithEmptyParamList<TReturnValue>
                 progress, cancel);
     }
 
-    // publicly visible Ice-internal type used to make request with input and output parameters,
-    // the input parameters are pass without the `in' modifier, this helper is used when there is a single
-    // input parameter and the parameter is of a non structure type.
+    /// <summary>Publicly visible Ice-internal type used to make request with input and output parameters, the input
+    /// parameters are pass without the `in' modifier, this helper is used when there is a single input parameter and
+    /// the parameter is of a non structure type.</summary>
```
Comment (externl) [inline csharp/src/Ice/OutgoingRequest.cs:99]:
```suggestion
    /// <summary>Publicly visible Ice-internal type used to make request with input and output parameters. The input
    /// parameters are passed without the `in' modifier. This helper is used when there is a single input parameter and
    /// the parameter is of a non structure type.</summary>
```

File: csharp/src/Ice/OutgoingRequest.cs [hunk 5]
```diff
@@ -138,10 +178,11 @@ public sealed class OutgoingRequestWithParam<TParamList, TReturnValue>
...
 
-    // publicly visible Ice-internal class used to make request with input and output parameters,
-    // the input parameters are pass with the `in' modifier, this helper is used when input parameter
-    // is an structure type, or when there is several input parameters as we pass them wrapped in a
-    // tuple.
+    /// <summary>Publicly visible Ice-internal class used to make request with input and output parameters, the input
+    /// parameters are pass with the `in' modifier, this helper is used when input parameter is an structure type, or
+    /// when there is several input parameters as we pass them wrapped in a tuple.</summary>
+    /// <typeparam name="TParamList">The input parameters type.</typeparam>
+    /// <typeparam name="TReturnValue">The return value type</typeparam>
```
Comment (externl) [inline csharp/src/Ice/OutgoingRequest.cs:185]:
```suggestion
    /// <summary>Publicly visible Ice-internal class used to make request with input and output parameters. The input
    /// parameters are passed with the `in' modifier. This helper is used when input parameter is an structure type, or
    /// when there is several input parameters as we pass them wrapped in a tuple.</summary>
    /// <typeparam name="TParamList">The input parameters type.</typeparam>
    /// <typeparam name="TReturnValue">The return value type</typeparam>
```

File: csharp/src/Ice/OutgoingRequest.cs [hunk 6]
```diff
@@ -274,16 +350,25 @@ public OutgoingRequestWithEmptyParamList(string operationName, bool idempotent,
                         cancel);
     }
 
-    // publicly visible Ice-internal class used to make void requests with input parameters, the input parameters
-    // are pass without the `in' modifier, this helper is used when there is a single input parameter and the parameter
-    // is not a structure type.
+    /// <summary>Publicly visible Ice-internal class used to make void requests with input parameters, the input
+    /// parameters are pass without the `in' modifier, this helper is used when there is a single input parameter and
+    /// the parameter is not a structure type.</summary>
+    /// <typeparam name="TParamList">The input parameters type.</typeparam>
```
Comment (externl) [inline csharp/src/Ice/OutgoingRequest.cs:356]:
```suggestion
    /// <summary>Publicly visible Ice-internal class used to make void requests with input parameters. The input
    /// parameters are passed without the `in' modifier. This helper is used when there is a single input parameter and
    /// the parameter is not a structure type.</summary>
    /// <typeparam name="TParamList">The input parameters type.</typeparam>
```

File: csharp/src/Ice/OutgoingRequest.cs [hunk 7]
```diff
@@ -334,9 +430,10 @@ public sealed class OutgoingRequestWithParam<TParamList> : OutgoingRequest
                         cancel);
     }
 
-    // publicly visible Ice-internal type used to make void request with input parameters,
-    // the input parameters are pass with the `in' modifier, this helper is used when input parameter
-    // is a structure type, or when there is several input parameters as we pass them wrapped in a tuple.
+    /// <summary>Publicly visible Ice-internal type used to make void request with input parameters, the input
+    /// parameters are pass with the `in' modifier, this helper is used when input parameter is a structure type, or
+    /// when there is several input parameters as we pass them wrapped in a tuple.</summary>
+    /// <typeparam name="TParamList">The input parameters type.</typeparam>
```
Comment (externl) [inline csharp/src/Ice/OutgoingRequest.cs:436]:
```suggestion
    /// <summary>Publicly visible Ice-internal type used to make void request with input parameters. The input
    /// parameters are passed with the `in' modifier. This helper is used when input parameter is a structure type, or
    /// when there is several input parameters as we pass them wrapped in a tuple.</summary>
    /// <typeparam name="TParamList">The input parameters type.</typeparam>
```

**Review by externl — APPROVED** [review]

File: csharp/src/Ice/IncomingFrame.cs [hunk 8]
```diff
@@ -14,6 +14,8 @@ namespace ZeroC.Ice
     /// <summary>Base class for incoming frames.</summary>
     public abstract class IncomingFrame
     {
+        /// <summary>The binary context is a collection of byte blobs with an associated int key, the dispatch and
+        /// invocation interceptors can use the binary context to marshall arbitrary data within a request.</summary>
```
Comment (bernardnormier) [inline csharp/src/Ice/IncomingFrame.cs:18]: marshal

File: csharp/src/Ice/Options.cs [hunk 9]
```diff
@@ -9,16 +9,21 @@
...
-        public enum State
+        private enum State
         {
             Normal,
             DoubleQuote,
             SingleQuote,
             ANSIQuote
         }
 
+        /// <summary>Split a string into a list of command line arguments, the string is parsed using semantics similar
```
Comment (bernardnormier) [inline csharp/src/Ice/Options.cs:23]: Splits

File: csharp/src/Ice/OutgoingRequest.cs [hunk 10]
```diff
@@ -62,12 +68,23 @@ public sealed class OutgoingRequestWithEmptyParamList<TReturnValue>
             _idempotent = idempotent;
         }
 
+        /// <summary>Invoke the request synchronously.</summary>
```
Comment (bernardnormier) [inline csharp/src/Ice/OutgoingRequest.cs:71]: Invokes

File: csharp/src/Ice/OutgoingRequestFrame.cs [hunk 11]
```diff
@@ -32,6 +32,7 @@ public sealed class OutgoingRequestFrame : OutgoingFrame
             }
         }
 
+        /// <summary>The request payload encoding.</summary>
```
Comment (bernardnormier) [inline csharp/src/Ice/OutgoingRequestFrame.cs:35]: The encoding of the request payload.

File: csharp/src/Ice/RemoteException.cs [hunk 12]
```diff
@@ -45,8 +48,11 @@ protected RemoteException()
         protected RemoteException(string? message, Exception? innerException)
             : base(message, innerException) => _hasCustomMessage = message != null;
 
-        // See InputStream
-        // This base implementation is only called on a plain RemoteException.
+        /// <summary>Unmarshal a remote exception from the <see cref="InputStream"/>. This base implementation is only
```
Comment (bernardnormier) [inline csharp/src/Ice/RemoteException.cs:51]: Unmarshals

File: csharp/src/Ice/RemoteException.cs [hunk 13]
```diff
@@ -45,8 +48,11 @@ protected RemoteException()
...
+        /// called on a plain RemoteException.</summary>
+        /// <param name="istr">The <see cref="InputStream"/> to read from.</param>
+        /// <param name="firstSlice"><c>True</c> if the exception correspond to the first Slice, <c>False</c>
```
Comment (bernardnormier) [inline csharp/src/Ice/RemoteException.cs:54]: corresponds

File: csharp/src/Ice/RemoteException.cs [hunk 14]
```diff
@@ -56,14 +62,18 @@ protected virtual void IceRead(InputStream istr, bool firstSlice)
 
         internal void Read(InputStream istr) => IceRead(istr, true);
 
-        // See OutputStream
-        // This implementation can only called on a plain RemoteException with IceSlicedData set.
+        /// <summary>Marshal a remote exception to the <see cref="OutputStream"/>. This implementation can only be
+        /// called on a plain RemoteException with IceSlicedData set.</summary>
+        /// <param name="ostr">The <see cref="OutputStream"/> to marshal the exception.</param>
+        /// <param name="firstSlice"><c>True</c> if the exception correspond to the first Slice, <c>False</c>
```
Comment (bernardnormier) [inline csharp/src/Ice/RemoteException.cs:68]: corresponds

File: csharp/src/Ice/StringUtil.cs [hunk 15]
```diff
@@ -8,20 +8,15 @@
 
 namespace ZeroC.Ice
 {
+    /// <summary>Helper methods for string manipulation</summary>
```
Comment (bernardnormier) [inline csharp/src/Ice/StringUtil.cs:11]: missing period.

**Review by bernardnormier — COMMENTED** [review]
Needs a few fixes.

**Pull request merged**